java/source/jdbc.rst

```diff
@@ -0,0 +1,309 @@
+   import java.sql.Connection;
+   import java.sql.DriverManager;
+   import java.sql.ResultSet;
+   import java.sql.SQLException;
+   import java.sql.Types;
+   import java.util.HashMap;
+
+   try (BufferAllocator allocator = new RootAllocator();
+        Connection connection = DriverManager.getConnection(
+                "jdbc:h2:mem:h2-jdbc-adapter")) {
+       ScriptRunner runnerDDLDML = new ScriptRunner(connection);
+       runnerDDLDML.setLogWriter(null);
+       runnerDDLDML.runScript(new BufferedReader(
+               new FileReader("./thirdpartydeps/jdbc/h2-ddl.sql")));
+       runnerDDLDML.runScript(new BufferedReader(
+               new FileReader("./thirdpartydeps/jdbc/h2-dml.sql")));
+       JdbcToArrowConfig config = new JdbcToArrowConfigBuilder(allocator,
+               JdbcToArrowUtils.getUtcCalendar())
+               .setTargetBatchSize(2)
+               .setArraySubTypeByColumnNameMap(
+                       new HashMap<>() {{
+                           put("LIST_FIELD19",
+                                   new JdbcFieldInfo(Types.INTEGER));
+                       }}
+               )
+               .build();
+       try (ResultSet resultSet = connection.createStatement().executeQuery(
+               "SELECT int_field1, bool_field2, bigint_field5, char_field16, list_field19 FROM TABLE1");
+            ArrowVectorIterator iterator = JdbcToArrow.sqlToArrowVectorIterator(
+                    resultSet, config)) {
+           while (iterator.hasNext()) {
+               try (VectorSchemaRoot root = iterator.next()) {
+                   System.out.print(root.contentToTSVString());
+               }
+           }
+       }
+   } catch (SQLException | IOException e) {
+       e.printStackTrace();
+   }
+
+.. testoutput::
+
+   INT_FIELD1    BOOL_FIELD2    BIGINT_FIELD5    CHAR_FIELD16    LIST_FIELD19
+   101    true    1000000000300    some char text      [1,2,3]
+   102    true    100000000030    some char text      [1,2]
+   INT_FIELD1    BOOL_FIELD2    BIGINT_FIELD5    CHAR_FIELD16    LIST_FIELD19
+   103    true    10000000003    some char text      [1]
+
+Configuring numeric (decimal) precision and scale
+=================================================
+
+By default, the scale of any decimal values must exactly match the defined
+scale of the Arrow type of the column, or else an UnsupportedOperationException
+will be thrown with a message like ``BigDecimal scale must equal that in the Arrow
+vector``will be thrown during conversion.
+
+This can happen because Arrow infers the type from the ResultSet metadata, which
+is not accurate for all database drivers. The JDBC adapter lets you avoid this
+by either overriding the decimal scale, or by providing a RoundingMode via
+``setBigDecimalRoundingMode`` to convert values to the expected scale.
+
+In this example, we have a BigInt column. By default, the inferred scale
+is 0. We override the scale to 7 and then set a RoundingMode to convert
+values to the given scale.
```

Comment:
```suggestion
By default, the scale of any decimal values must exactly match the defined
scale of the Arrow type of the column, or else an UnsupportedOperationException
will be thrown with a message like ``BigDecimal scale must equal that in the Arrow
vector``.

This can happen because Arrow infers the type from the ResultSet metadata, which
is not accurate for all database drivers. The JDBC adapter lets you avoid this
by either overriding the decimal scale, or by providing a RoundingMode via
``setBigDecimalRoundingMode`` to convert values to the expected scale.

In this example, we have a BigInt column. By default, the inferred scale
is 0. We override the scale to 7 and then set a RoundingMode to convert
values to the given scale.
```